ui/types/api/dialog.d.ts

```diff
@@ -56,9 +51,14 @@ export type QDialogInputPrompt<T = string> = {
    */
   isValid?: (value: T) => boolean;
 } & Partial<Omit<QInputProps, "modelValue" | "onUpdate:modelValue" | "type">> &
-  Partial<NonFunctionProperties<HTMLInputElement>>;
+  Omit<InputHTMLAttributes, "type"> & {
+    // Has type-checking but no autocompletion: https://github.com/microsoft/TypeScript/issues/41620
+    [dataAttributes in `data-${string}`]?: any;
+  } & {
+    "data-cy"?: string;
+  };
```

Comment:
Accepting any attribute was too broad, could lead to typos and such. Aside from the element-specific attributes, common attributes like `aria-*` was already being handled. So, pretty much the other thing left is `data-*`. Because of that, I converted it to only allow `data-*`, then on top of that added `data-cy` specifically for ease of access via auto-completion.